host-configs/corona-toss_4_x86_64_ib-clang@13.0.0_rocm.cmake

```diff
@@ -0,0 +1,113 @@
+# !!!! This is a generated file, edit at own risk !!!!
+#------------------------------------------------------------------------------
+# CMake executable path: /usr/tce/packages/cmake/cmake-3.21.1/bin/cmake
+#------------------------------------------------------------------------------
+
+#------------------------------------------------------------------------------
+# Compilers
+#------------------------------------------------------------------------------
+# Compiler Spec: clang@13.0.0
+#------------------------------------------------------------------------------
+if(DEFINED ENV{SPACK_CC})
+
+  set(CMAKE_C_COMPILER "/usr/WS1/axom/libs/toss_4_x86_64_ib/2022_03_09_10_28_32/spack/lib/spack/env/clang/clang" CACHE PATH "")
+
+  set(CMAKE_CXX_COMPILER "/usr/WS1/axom/libs/toss_4_x86_64_ib/2022_03_09_10_28_32/spack/lib/spack/env/clang/clang++" CACHE PATH "")
+
+  set(CMAKE_Fortran_COMPILER "/usr/WS1/axom/libs/toss_4_x86_64_ib/2022_03_09_10_28_32/spack/lib/spack/env/clang/flang" CACHE PATH "")
+
+else()
+
+  set(CMAKE_C_COMPILER "/usr/tce/packages/rocmcc-tce/rocmcc-4.5.2/bin/amdclang" CACHE PATH "")
+
+  set(CMAKE_CXX_COMPILER "/usr/tce/packages/rocmcc-tce/rocmcc-4.5.2/bin/hipcc" CACHE PATH "")
+
+  set(CMAKE_Fortran_COMPILER "/usr/tce/packages/rocmcc-tce/rocmcc-4.5.2/bin/amdflang" CACHE PATH "")
+
+endif()
+
+set(CMAKE_Fortran_FLAGS "-Mfreeform" CACHE STRING "")
+
+set(ENABLE_FORTRAN ON CACHE BOOL "")
+
+#------------------------------------------------------------------------------
+# MPI
+#------------------------------------------------------------------------------
+
+set(MPI_C_COMPILER "/usr/tce/packages/mvapich2/mvapich2-2.3.6-gcc-10.2.1/bin/mpicc" CACHE PATH "")
+
+set(MPI_CXX_COMPILER "/usr/tce/packages/mvapich2/mvapich2-2.3.6-gcc-10.2.1/bin/mpicxx" CACHE PATH "")
+
+set(MPI_Fortran_COMPILER "/usr/tce/packages/mvapich2/mvapich2-2.3.6-gcc-10.2.1/bin/mpif90" CACHE PATH "")
+
+set(MPIEXEC_EXECUTABLE "/usr/bin/srun" CACHE PATH "")
+
+set(MPIEXEC_NUMPROC_FLAG "-n" CACHE STRING "")
+
+set(ENABLE_MPI ON CACHE BOOL "")
+
+#------------------------------------------------------------------------------
+# Hardware
+#------------------------------------------------------------------------------
+
+#------------------------------------------------------------------------------
+
+# HIP
+
+#------------------------------------------------------------------------------
+
+
+set(ENABLE_HIP ON CACHE BOOL "")
+
+set(HIP_ROOT_DIR "/opt/rocm-4.5.2/hip" CACHE STRING "")
+
+set(HIP_CLANG_PATH "/opt/rocm-4.5.2/hip/../llvm/bin" CACHE STRING "")
+
+set(CMAKE_HIP_ARCHITECTURES "gfx906" CACHE STRING "")
+
+set(CMAKE_EXE_LINKER_FLAGS "-Wl,--disable-new-dtags -L/opt/rocm-4.5.2/hip/../llvm/lib -L/opt/rocm-4.5.2/hip/lib -Wl,-rpath,/opt/rocm-4.5.2/hip/../llvm/lib:/opt/rocm-4.5.2/hip/lib -lpgmath -lflang -lflangrti -lompstub -lamdhip64 -L/opt/rocm-4.5.2/hip/../lib64 -Wl,-rpath,/opt/rocm-4.5.2/hip/../lib64 -lhsakmt -lamd_comgr" CACHE STRING "")
```

Comment:
Have you tried limiting these?  This seems extremely excessive for a user to have to add to the link line.  I would have expected CMake to auto find most of this or LC to wrap them in a wrapper.

Reply:
Is it possibly because this is the preview machine and will be fixed later?

Reply:
@white238 FWIW, this link line is basically the same as what we do for RAJA. According to John G., this is the only way to link with the ROCm compilers "in a sane way"....

Reply:
@white238 yes, I've tried cutting this down, but it's been necessary for things to link. John Gyllenhaal has this link line documented as part of the rznevada quickstart guide on Confluence.

Reply:
Sigh... okay I should also provide an example in BLT i guess.